intercom_tag_users_block.py

```diff
@@ -0,0 +1,46 @@
+from nio.block.base import Block
+from nio.signal.base import Signal
+from nio.properties import IntProperty, StringProperty, ObjectProperty, \
+    PropertyHolder
+import requests
+
+
+class IntercomTagUsers(Block):
+
+    access_token = StringProperty(
+        title="Access Token", default="[[INTERCOM_ACCESS_TOKEN]]")
+    tag_name = StringProperty(title="Name of Tag", default="CrankyCustomer")
+    email = StringProperty(title="User Email", default="{{ $user }}")
+    user_id = StringProperty(
+        title="User ID", default="{{ $user_id }}", allow_none=True)
+
+    def configure(self, context):
+        super().configure(context)
+
+    def process_signals(self, signals):
+        for signal in signals:
+            response = self._request('post', body={
+                "name": self.tag_name(signal),
+                "users": [
+                    {"id": self.user_id(signal)},
+                    {"email": self.email(signal)}
+                ] if self.user_id(signal) else [{"email": self.email(signal)}],
+            })
+            if response.status_code != 200:
+                raise Exception
+
+    def _request(self, method='post', id=None, body=None):
+        url = 'https://api.intercom.io/tags'
+        kwargs = {}
+        kwargs['headers'] = {
+            "Authorization": "Bearer {}".format(self.access_token()),
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+        }
+        kwargs['json'] = body
+        response = getattr(requests, method)(url, **kwargs)
```

Comment:
I'm guessing this is copy/pasted from the other block. In this case, it might look a little more simple here if you just called `requests.post` here and remove the `method` argument from this function.